EIPS/eip-6190.md

```diff
@@ -0,0 +1,51 @@
+---
+eip: 6190
+title: Functional SELFDESTRUCT
+description: Changes SELFDESTRUCT to only cause a finite number of state changes
+author: Pandapip1 (@Pandapip1), Alex Beregszaszi (@axic)
+discussions-to: https://ethereum-magicians.org/t/eip-6190-functional-selfdestruct/12232
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-12-20
+requires: 6189
+---
+
+## Abstract
+
+Changes `SELFDESTRUCT` to only cause a finite number of state changes.
+
+## Motivation
+
+The `SELFDESTRUCT` instruction has a fixed price, but is unbounded in storage/account changes (it needs to delete all keys). This has been an outstanding concern for some time.
+
+Furthermore, *Verkle trees* might accounts will be organised differently. Account properties, including storage, would have individual keys. It would not be possible to traverse and find all used keys. This makes `SELFDESTRUCT` very challenging to support in Verkle trees. This EIP is a step towards supporting `SELFDESTRUCT` in Verkle trees.
```

Comment:
```suggestion
Furthermore, with *Verkle trees* accounts will be organised differently. Account properties, including storage, would have individual keys. It would not be possible to traverse and find all used keys. This makes `SELFDESTRUCT` very challenging to support in Verkle trees. This EIP is a step towards supporting `SELFDESTRUCT` in Verkle trees.
```